chia/wallet/transaction_sorting.py

```diff
@@ -0,0 +1,14 @@
+import enum
+
+
+class SortKey(str, enum.Enum):
+    # "?" will be mapped to either ASC or DESC (when reverse is set to true)
+    CONFIRMED_AT_HEIGHT = "order by confirmed_at_height ASC"
+    RELEVANCE = "order by confirmed ASC, confirmed_at_height DESC, created_at_time DESC"
+
+    @staticmethod
+    def reverse(query: str) -> str:
+        reversed_query = query.replace("ASC", "TEMP")
+        reversed_query = reversed_query.replace("DESC", "ASC")
+        reversed_query = reversed_query.replace("TEMP", "DESC")
```

Comment:
It's not clear to me what you gain from replacing `ASC` -> `TEMP` and then `TEMP` -> `ASC`.
It seems risky to reverse the sort order this way though. What if the query happens to contain the string `TEMP`?
I think it would be much safer (and possibly more straight-forward) to decide between `ASC` and `DESC` when we build the query string instead.

Reply:
I wish it didn't look as hacky as it did but I couldn't see another way.  It goes `ASC` -> `TEMP`, `DESC -> ASC` and then `TEMP -> `DESC`.  The method is only meant to be run on the enum values above it, although I suppose that could be clearer.  For the enum value to contain the string `TEMP` would be very unlikely considering it would have to be a column name that we are sorting on.

Reply:
Without saying I like this, the following gets rid of a couple smells.  Maybe the method names should change.

```python
class SortKey(str, enum.Enum):
    CONFIRMED_AT_HEIGHT = "order by confirmed_at_height {ASC}"
    RELEVANCE = "order by confirmed {ASC}, confirmed_at_height {DESC}, created_at_time {DESC}"

    def ascending(self) -> str:
        return self.value.format(ASC="ASC", DESC="DESC")

    def descending(self) -> str:
        return self.value.format(ASC="DESC", DESC="ASC")

print(SortKey.CONFIRMED_AT_HEIGHT.ascending())
print(SortKey.CONFIRMED_AT_HEIGHT.descending())

print(SortKey.RELEVANCE.ascending())
print(SortKey.RELEVANCE.descending())
```

```
order by confirmed_at_height ASC
order by confirmed_at_height DESC
order by confirmed ASC, confirmed_at_height DESC, created_at_time DESC
order by confirmed DESC, confirmed_at_height ASC, created_at_time ASC
```

Reply:
Actually, probably better to just not inherit from `str` with that proposal.